pkg/runtime/embedded.go

```diff
@@ -24,9 +24,9 @@ import (
 )
 
 type encodable struct {
-	e        Encoder `json:"-"`
+	E        Encoder `json:"-"`
 	obj      Object
-	versions []unversioned.GroupVersion `json:"-"`
+	Versions []unversioned.GroupVersion `json:"-"`
```

Comment:
No. Just remove the json tag completely. Those tags are trying to tell the JSON package not to marshal or unmarshal those fields, but the JSON package already ignores private members.